hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
> So what about:

I like this implementation. I'll be committing a change shortly to match it. I did find in my testing however, that even with a threshold of `n<200`, plotting the bare histogram is not very useful. 

Here are some plots from an EELS spectrum image with `n = 164`. Are these more useful than the "max" histogram? It's hard for me to say.

![histogram with bins blocks](https://cloud.githubusercontent.com/assets/1278301/8986625/00b3a8bc-36ab-11e5-9ac4-2be7efce0630.png)
![histogram with bins freedman](https://cloud.githubusercontent.com/assets/1278301/8986629/00b6da0a-36ab-11e5-88a2-0c1dca84d22d.png)
![histogram with bins scotts](https://cloud.githubusercontent.com/assets/1278301/8986630/00b6f6fc-36ab-11e5-91f9-6335837ee21b.png)
![histogram with bins knuth](https://cloud.githubusercontent.com/assets/1278301/8986627/00b566d4-36ab-11e5-90f0-db9ab012ff92.png)
![histogram with bins 500](https://cloud.githubusercontent.com/assets/1278301/8986628/00b59f1e-36ab-11e5-948a-b8bc6e5f5afc.png)
![histogram with bins 100](https://cloud.githubusercontent.com/assets/1278301/8986626/00b51238-36ab-11e5-98bd-ca866a0331ce.png)